docs: add custom objects beta API documentation (#96)
devrev/fern-api-docs: fern/docs/pages/custom-objects.mdx

```
curl --location 'https://api.devrev.ai/custom-objects.update' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'Authorization: <YOUR_API_TOKEN>' \
--data '{
    "leaf_type": "campaign",
    "id": "<OBJECT_ID>",
    "custom_fields": {
        "budget": 15000
    }
}'
```

### Delete a Custom Object

To delete a custom object, use the `custom-objects.delete` endpoint:

```
curl --location 'https://api.devrev.ai/custom-objects.delete' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'Authorization: <YOUR_API_TOKEN>' \
--data '{
    "id": "<OBJECT_ID>"
}
```

## Key Concepts

1. **Leaf Type**: The base type of your custom object (e.g., "campaign").
2. **Subtype**: A more specific categorization within a leaf type (e.g., "promotion" or "advertising" for a "campaign" leaf type).
3. **Custom Fields**: User-defined fields that store specific data for your custom object.
4. **Unique Key**: A unique identifier for each custom object, useful for maintaining idempotency.

> For more details, please refer to the [Customization](./customization.mdx) documentation.

Author: @Atul-Butola please let me know if this isn't the right way to link other docs.

Author: I tested this on the published docs preview and it doesn't work.

Author: Fixed this.

## Table of contents

> * [Introduction](#custom-objects)

Reviewer: Remove. Fern automatically constructs a TOC from the headings so no need to create one explicitly.

* [Quick start](#quick-start)
   - [Create a schema and a custom object](#create-a-schema-and-a-custom-object)
   - [Get a custom object](#get-a-custom-object)
   - [List custom objects](#list-custom-objects)
   - [Update a custom object](#update-a-custom-object)
   - [Delete a custom object](#delete-a-custom-object)
* [Key concepts](#key-concepts)
* [Custom object lifecycle](#custom-object-lifecycle)
* [Extended use cases](#extended-use-cases)
    - [Reference custom objects in other objects](#reference-custom-objects-in-other-objects)
    - [Add subtypes to custom objects](#add-subtypes-to-custom-objects)

- - -

> ## Introduction

Reviewer: Remove.
```suggestion
```

Custom objects allow you to extend DevRev's data model beyond the standard use-cases
served by the native apps like Build and Support.  Custom objects allow you to create
and manage object types tailored to your specific business needs.

> ## Quick start

Reviewer: ```suggestion
## Custom object example
```

Author: How about "Getting started"? "Custom object example" doesn't fit well with the flow of the doc.

Let's say you want to manage marketing campaigns on DevRev. We'll go through the process of creating a "Campaign" custom object with relevant fields.

All DevRev objects require a schema. So, we'll first create a schema for the "Campaign" custom object. Make sure to replace the `<YOUR_API_TOKEN>` with your actual API token.

> ### Create a schema and a custom object

Reviewer: Insert an H2 here
```suggestion
## Custom object lifecycle

### Create a schema and a custom object
```

```
curl --location 'https://api.devrev.ai/schemas.custom.set' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'Authorization: <YOUR_API_TOKEN>' \
--data '{
    "type": "tenant_fragment",
    "description": "Attributes for Campaign",
    "leaf_type": "campaign",
    "fields": [
        {
            "name": "name",
            "field_type": "tokens",
            "description": "Name of the campaign"
        },
        {
            "name": "start_date",
            "field_type": "date",
            "description": "Start date of the campaign"
        },
        {
            "name": "end_date",
            "field_type": "date",
            "description": "End date of the campaign"
        },
        {
            "name": "budget",
            "field_type": "int",
            "description": "Budget allocated for the campaign"
        },
        {
            "name": "target_audience",
            "field_type": "enum",
            "description": "Target audience for the campaign",
            "allowed_values": [
                "Professionals",
                "Students"
            ]
        }
    ],
    "is_custom_leaf_type": true,

>     "id_prefix": "CAMP"

Reviewer 2: Can we add definitions regarding `is_custom_leaf_type` and `id_prefix`?

--header 'Authorization: <YOUR_API_TOKEN>' \
--data '{
    "leaf_type": "campaign",
    "filter": [
      "eq",
      "$custom_fields.tnt__target_audience",
      "Young adults"
    ]
}'
```

### Update a custom object

To update an existing custom object, use the `custom-objects.update` endpoint. The
following example updates the budget of the previously created campaign custom object:

```
curl --location 'https://api.devrev.ai/custom-objects.update' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'Authorization: <YOUR_API_TOKEN>' \
--data '{
    "id": "don:core:dvrv-us-1:devo/demo:custom_object/campaign/1",
    "custom_fields": {
        "tnt__budget": 15000
    }
}'
```

### Delete a custom object

To delete a custom object, use the `custom-objects.delete` endpoint.
The following example deletes the previously created campaign custom object:

```
curl --location 'https://api.devrev.ai/custom-objects.delete' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'Authorization: <YOUR_API_TOKEN>' \
--data '{
    "id": "don:core:dvrv-us-1:devo/demo:custom_object/campaign/1"
}'
```

> ## Key concepts

Reviewer: Move this up to be the first section after the intro lines.

> 1. **Leaf Type**: The base type of your custom object (e.g., "campaign"). Note that for custom objects, `is_custom_leaf_type` must be set to `true` to differentiate it from standard DevRev objects.

Reviewer: ```suggestion
1. **Leaf type**: The base type of your custom object (for example, "campaign"). For custom objects, `is_custom_leaf_type` must be set to `true` to differentiate it from standard DevRev objects.
```

> 2. **Subtype**: A more specific categorization within a leaf type (e.g., "promotion" or "advertising" for a "campaign" leaf type).

Reviewer: ```suggestion
2. **Subtype**: A more specific categorization within a leaf type (for example, "promotion" or "advertising" for a "campaign" leaf type).
```

> 3. **Custom Fields**: User-defined fields that store specific data for your custom object.

Reviewer: ```suggestion
3. **Custom fields**: User-defined fields that store specific data for your custom object.
```

> 4. **Unique Key**: The `unique_key` is a unique identifier for each custom object. This is useful for maintaining idempotency when retrying custom object creation requests.

Reviewer: ```suggestion
4. **Unique key**: A unique identifier for each custom object. This is useful for maintaining idempotency when retrying custom object creation requests.
```

> 5. **ID Prefix**: The `id_prefix` should be unique and is used to generate the display ID for the custom object. In the above example, the `id_prefix` is "CAMP" and the generated custom object display ID is "C-CAMP-1". The display ID is used to identify the custom object in the UI, similar to the standard DevRev object display IDs like "ISS-001" for issues.

Reviewer: ```suggestion
5. **ID prefix**: Used to generate the display ID for the custom object. In the above example, the `id_prefix` is "CAMP" and the generated custom object display ID is "C-CAMP-1". The display ID is used to identify the custom object in the UI, similar to the standard DevRev object display IDs like "ISS-001" for issues.
```

For more details on customization concepts, please refer to the
[Customization](./object-customization) documentation.

> ## Custom object lifecycle

Reviewer: Move the contents of this section up to the lifecycle operations reference.

1. **Creation**: Define the leaf type, subtype (optional), and custom fields for your custom object.
2. **Usage**: Create, update, and query custom objects as needed in your workflows.
3. **Management**: Modify the schema of the custom object as your needs evolve.

> ## Extended use cases

Reviewer: Is there another description you could use besides "extended"?

Author: How about "Beyond basic use cases".
Some other alternatives:
1. Advanced Scenarios
1. Advanced Applications
1. Complex Use Cases
1. Advanced Implementations
1. Specialized Configurations
1. Advanced Features
1. Power User Techniques
1. Advanced Customization
1. Beyond Basic Usage
1. Advanced Functionality

Reviewer: Extended, advanced, and beyond basic all have the same meaning, which is nothing. So if there isn't a term that's meaningful I guess you can leave it as-is.
